src/init.c

```diff
@@ -732,6 +732,15 @@ void _julia_init(JL_IMAGE_SEARCH rel)
 
     jl_gc_enable(0);
 
+    // We need `gcstack` in `Task` to allocate Julia objects; *including* the `Task` type.
+    // However, to allocate a `Task` via `jl_gc_alloc` as done in `jl_init_root_task`,
+    // we need the `Task` type itself. We use stack-allocated "raw" `jl_task_t` struct to
+    // workaround this chicken-and-egg problem. Note that this relies on GC to be turned
+    // off just above as GC fails because we don't/can't allocate the type tag.
+    jl_task_t bootstrap_task;
+    bootstrap_task.gcstack = NULL;
+    jl_current_task = &bootstrap_task;
```

Comment:
This is what I suggest for solving the bootstrapping issue I mentioned in the OP. It's a bit scary approach but I find it the most straightforward way to do it.

Also, do we want to zero out the struct?